jina/serve/runtimes/worker/batch_queue.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+from asyncio import Event
+from typing import List, Dict, Any
+
+from jina import DocumentArray
+from jina.serve.executors import BaseExecutor
+from jina.types.request.data import DataRequest
+
+# TODO: Make sure we dont deserialize the request twice
+# TODO: SIGINT/SIGTEM: Flush
+
+async def sleep_then_set(time_seconds: int, event: Event):
+    """Sleep for time_seconds and then set the event"""
+    asyncio.sleep(time_seconds)
+    event.set()
+
+class BatchQueue():
+    """A batch queue that holds the data request and the executor."""
+
+    def __init__(self, executor: BaseExecutor, exec_endpoint: str, args: Any, preferred_batch_size: int=4, timeout: int=10_000) -> None:
+        self._preferred_batch_size: int = preferred_batch_size
+        self._timeout: int = timeout
+        self._executor: BaseExecutor = executor
+        self._exec_endpoint: str = exec_endpoint
+        self.args = args
+
+        self._reset()
+    
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}({self._executor}, {self._exec_endpoint}, preferred_batch_size={self._preferred_batch_size}, timeout={self._timeout})'
+    
+    def __str__(self) -> str:
+        self.__repr__()
+    
+    def _reset(self) -> None:
+        """Set all events and reset the batch queue."""
+        if getattr(self, '_events', None) is not None:
+            [i.set() for i in self._events]
+        self._events: List[Event] = []
+        self._requests: List[DataRequest] = []
+        self._request_lens: List[int] = []
+        self._big_doc: DocumentArray = DocumentArray.empty()
+
+        self._flush_trigger: Event = Event()
```

Comment:
can we initialize these (also) directly in the `__init__()`? I like it when all the state is explicitly in the init, makes it easier to know what fields exist on this object.

Reply:
Yea maybe we should just duplicate some of the code so that concern 1 moves into `__init__`